### PR TITLE
Add mimetype for ReStructuredText

### PR DESCRIPTION
**Pull request by kyteinsky**
## Summary
Mimetype for re structured text. It is aliased to text.  
There is no official mimetype for `.rst` files so `text/plain` can also work here.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits

### CODE REVIEW

**Comment by kesselb** [comment]
If you want to update the mime type for files already existing in oc_filecache an additional change for `lib/private/Repair/RepairMimeTypes.php` is necessary.

Example: https://github.com/nextcloud/server/pull/41395/files#diff-f89544ac76d4f0e2cd70b4979d625b3992a72fa26c6f0b5fe92ec0364b9716e9

**Comment by kesselb** [comment]
> I had assumed repair was a file preview thing. Added it now, thanks for the explanation. Do the changes in `tests/data/integritycheck/mimetypeListModified/core/js/mimetypelist.js` also need to be pushed?

Yep.

There should be a brief inline description in CheckerTest how to update it properly.

**Comment by kyteinsky** [comment]
The CheckerTest did not update the `core/js/mimetypelist.js` file, which was updated with `./occ maintenance:mimetype:update-js` as mentioned in the file.

File: core/js/mimetypelist.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ OC.MimeTypeList={
     "application/font-sfnt": "font",
     "application/font-woff": "font",
     "application/gpx+xml": "location",
+    "application/gzip": "package/x-generic",
```
Comment (kesselb) [inline core/js/mimetypelist.js:16]: Would you mind double-checking this alias?
It looks wrong/unrelated to me.

Comment (kyteinsky) [reply core/js/mimetypelist.js:16]: It was put there by `./occ maintenance:mimetype:update-js`. It still is the case. Maybe someone forgot to run the update command or removed it manually?  
It looks in sync with `mimetypealiases.dist.json` file.

Comment (kesselb) [reply core/js/mimetypelist.js:16]: Thanks for checking :+1:

**Review by kesselb — APPROVED** [review]

File: version.php [hunk 2]
```diff
@@ -30,7 +30,7 @@
 // between betas, final and RCs. This is _not_ the public version number. Reset minor/patch level
 // when updating major/minor version number.
 
-$OC_Version = [29, 0, 0, 5];
+$OC_Version = [29, 0, 0, 6];
```
Comment (emoral435) [inline version.php:33]: Is this new version update really required?

**Review by emoral435 — COMMENTED** [review]
AFAIK (and I very well can be wrong, so correct me if I am), this should not cause any problems with how we handle mimetyping in other parts of our frontend 👍 So if this version upgrade is really needed, then this looks good to me!

File: version.php (hunk 2 above)
Comment (skjnldsv) [reply version.php:33]: I think it does, in any case, since we're releasing 29, we can leave the upgrade in i'd say :thinking:

**Comment by skjnldsv** [comment]
Conflicts @kyteinsky

File: version.php (hunk 2 above)
Comment (kyteinsky) [reply version.php:33]: yes, it is needed for the repair step to re-label the mimetypes of `.rst` files

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**